### DISCUSSION TITLE
Escape {0}, {1}, {N}... in commands and files strings

### DISCUSSION THREAD

**Discussion by ghost** [evilmartians/lefthook#763] [Q&A] [answered]
I have a use case where I need to use `HEAD@{1}` in the `files` property of the config, but the `{1}` is being replaced with the first argument of the relevant git hook. In my case, this is `0` as the hook is `post-merge`, and this wasn't a rebase (`1` if it was).

Is there a way to escape the `{1}` here so it isn't interpolated? 

Apologies if this has been answered before; I could not find any relevant Q&As / issues about it

**Comment by ghost** [comment, marked as answer]
As an aside, I've worked around this for now with a little shell magic:

```shell
git diff-tree -r --name-only --no-commit-id HEAD@{$(echo 1)} HEAD
```

Looking at the relevant code it seems like there isn't any built-in way to do this

https://github.com/evilmartians/lefthook/blob/c15e15c263caf60e69a689f6060c0275824abed4/internal/lefthook/runner/prepare_command.go#L183-L189

**Reply by mrexox** [reply]
Hey! There was a similar issue: https://github.com/evilmartians/lefthook/issues/479. Maybe its workaround may help you (using `HEAD^1` instead of `HEAD@{1}`)

**Reply by ghost** [reply]
I've been scouring the `git` docs (a hard read, honestly!), and my take is that `HEAD^1` isn't the same as `HEAD@{1}`.

* `HEAD@{1}` - references the place from where you jumped to new commit location based on `git reflog`
* `HEAD^1` - `^` revision parameter means the first parent of that commit object

I'm not well versed enough in `git` to say if these refer to the same commit _in this scenario_, but thought it worth a mention!

**Reply by mrexox** [reply]
That's true! This is not a fair replacement but it works in most cases :)
